False negative `unnecessary-list-index-lookup` with `iterable` kwarg

### Bug description

While there are plenty of tests for unnecessary_list_index_lookup, looks like missing is the case for using `iterable` as a kwarg

```
# test.py
series = [1, 2, 3]
for idx, val in enumerate(iterable=series, start=0):
    print(series[idx])  # should report [unnecessary-list-index-lookup] but currently isn't
```

Same code with iterable as an arg correctly reports and is tested.

### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```


### Expected behavior

`test.py:4:10: R1736: Unnecessary list index lookup, use 'val' instead (unnecessary-list-index-lookup)`

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.12
Python 3.11.0
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7770.false_negative
+++ b/doc/whatsnew/fragments/7770.false_negative
@@ -1,0 +1,3 @@
+Fixes ``unnecessary-list-index-lookup`` false negative when ``enumerate`` is called with ``iterable`` as a kwarg.
+
+Closes #7770

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -2146,9 +2146,17 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             not isinstance(node.iter, nodes.Call)
             or not isinstance(node.iter.func, nodes.Name)
             or not node.iter.func.name == "enumerate"
-            or not node.iter.args
-            or not isinstance(node.iter.args[0], nodes.Name)
         ):
+            return
+
+        try:
+            iterable_arg = utils.get_argument_from_call(
+                node.iter, position=0, keyword="iterable"
+            )
+        except utils.NoSuchArgumentError:
+            return
+
+        if not isinstance(iterable_arg, nodes.Name):
             return
 
         if not isinstance(node.target, nodes.Tuple) or len(node.target.elts) < 2:
@@ -2166,7 +2174,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             # is not redundant, hence we should not report an error.
             return
 
-        iterating_object_name = node.iter.args[0].name
+        iterating_object_name = iterable_arg.name
         value_variable = node.target.elts[1]
 
         # Store potential violations. These will only be reported if we don't

--- a/tests/functional/u/unnecessary/unnecessary_list_index_lookup.py
+++ b/tests/functional/u/unnecessary/unnecessary_list_index_lookup.py
@@ -130,3 +130,11 @@ def return_start(start):
 
 for i, k in enumerate(series, return_start(20)):
     print(series[idx])
+
+for idx, val in enumerate(iterable=series, start=0):
+    print(series[idx])  # [unnecessary-list-index-lookup]
+
+result = [my_list[idx] for idx, val in enumerate(iterable=my_list)]  # [unnecessary-list-index-lookup]
+
+for idx, val in enumerate():
+    print(my_list[idx])

--- a/tests/functional/u/unnecessary/unnecessary_list_index_lookup.txt
+++ b/tests/functional/u/unnecessary/unnecessary_list_index_lookup.txt
@@ -6,3 +6,5 @@ unnecessary-list-index-lookup:112:10:112:21::Unnecessary list index lookup, use 
 unnecessary-list-index-lookup:115:10:115:21::Unnecessary list index lookup, use 'val' instead:HIGH
 unnecessary-list-index-lookup:119:10:119:21::Unnecessary list index lookup, use 'val' instead:INFERENCE
 unnecessary-list-index-lookup:122:10:122:21::Unnecessary list index lookup, use 'val' instead:INFERENCE
+unnecessary-list-index-lookup:135:10:135:21::Unnecessary list index lookup, use 'val' instead:HIGH
+unnecessary-list-index-lookup:137:10:137:22::Unnecessary list index lookup, use 'val' instead:HIGH

